File: go/ManifoldApi/GetUser.go
package ManifoldApi

import (
	"encoding/json"
	"fmt"
)

type UserProfitCached struct {
	AllTime float64 `json:"allTime"`
	Monthly float64 `json:"monthly"`
	Daily   float64 `json:"daily"`
	Weekly  float64 `json:"weekly"`
}
type User struct {
	ID                   string           `json:"id"`
	CreatedTime          int64            `json:"createdTime"`
	Name                 string           `json:"name"`
	Username             string           `json:"username"`
	Balance              float64          `json:"balance"`
	TotalDeposits        float64          `json:"totalDeposits"`
	ProfitCached         UserProfitCached `json:"profitCached"`
	IsBot                bool             `json:"isBot"`
	IsAdmin              bool             `json:"isAdmin"`
	IsTrustworthy        bool             `json:"isTrustworthy"`
	CurrentBettingStreak int              `json:"currentBettingStreak"`
	LastBetTime          int64            `json:"lastBetTime"`
}

func GetUser(userId string) User {
	sb := callManifoldApi("GET", fmt.Sprintf("v0/user/by-id/%s", userId), nil)

	var response User
	json.Unmarshal([]byte(sb), &response)
	return response
}

File: src/api/getUser.ts
import z from "zod";
import { callManifoldApi } from "./callManifoldApi.ts";

export const UserSchema = z.object({
  id: z.string(),
  createdTime: z.number(),
  name: z.string().optional(),
  username: z.string().optional(),
  balance: z.number().optional(),
  totalDeposits: z.number().optional(),
  isBot: z.boolean().optional(),
  isAdmin: z.boolean().optional(),
  isTrustworthy: z.boolean().optional(),
  currentBettingStreak: z.number().optional(),
  lastBetTime: z.number().optional(),
});

export type User = z.infer<typeof UserSchema>;

export async function getUser(userId: string): Promise<User> {
  const sb = await callManifoldApi("GET", `v0/user/by-id/${userId}`);
  return UserSchema.parse(JSON.parse(sb));
}
